lib/galaxy/datatypes/data.py

```diff
@@ -1212,6 +1212,51 @@ def regex_line_dataprovider(
 class Directory(Data):
     """Class representing a directory of files."""
 
+    file_ext = "directory"
+
+    # The behavior of this class is intended to be similar to that of
+    # composite types, but with arbitrary structure of extra_files.
+
+    # Directories converted from archives, and possibly others, have a single
+    # root folder inside the directory. The root_folder attribute lets tools
+    # access that folder without first exploring the directory tree themselves.
+    # Will be set to the empty string for flat directories.
+    MetadataElement(
```

Comment:
Can we remove this ? The root directory should just be `$input.extra_files_path`

Reply:
Sure. I'm trying to remember why this was needed. I think it was because some zip files might compress the "target directory" or a "container directory" and we wanted to get the "target directory" in any case, but I see this should not have anything to do with the "base" Directory datatype and probably it should go into a sub-class when needed :+1: